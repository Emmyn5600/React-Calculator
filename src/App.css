.display-value {
  width: 300px;
  height: 80px;
  margin-left: auto;
  margin-right: auto;
  border: 1px solid gray;
  border-radius: 2px;
}

.board {
  width: 300px;
  margin-left: auto;
  margin-right: auto;
}

button {
  width: 25%;
  height: 50px;
  border-radius: 0;
  border: 1px solid gray;
}

#result {
  margin-left: 520px;
  width: 22.5%;
}