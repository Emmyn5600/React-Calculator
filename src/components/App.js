/* eslint-disable react/no-access-state-in-setstate */
import React from 'react';
import Display from './Display';
import ButtonPannel from './ButtonPanel';
import calculate from '../logic/calculator';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      operation: null,
    };

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(buttonName) {
    const calculation = calculate(buttonName, this.state);
    this.setState({
      total: calculation.total,
      next: calculation.next,
      operation: calculation.operation,
    });
  }

  render() {
    const { total, next } = this.state;
    return (
      <>
        <Display calculation={next || total} />
        <ButtonPannel clickHandler={(e) => this.handleClick(e.target.innerText)} />
      </>
    );
  }
}
export default App;
