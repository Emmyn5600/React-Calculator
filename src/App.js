import './App.css';

function App() {
  return (
    <div className="FirstApp">
      <h1>Hello world!!!</h1>
    </div>
  );
}

export default App;
