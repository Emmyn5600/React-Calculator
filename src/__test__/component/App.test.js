import React from 'react';
import renderer from 'react-test-renderer';
import App from '../../components/App';

describe('App render', () => {
  it('should render calculator', () => {
    const tree = renderer.create(<App/>).toJSON();
    expect(tree).toMatchSnapshot(); 
   })
});