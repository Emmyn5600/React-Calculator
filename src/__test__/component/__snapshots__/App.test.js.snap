// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render renders the App 1`] = `
<div
  className="app"
>
  <div
    className="lets"
  >
    <h1>
      Lets Do Some Magicians with Math Calc! 
    </h1>
  </div>
  <div
    className="calc"
  >
    <div
      className="display-result"
    >
      <input
        defaultValue={null}
        dir="rtl"
        id="result"
        placeholder="0"
      />
    </div>
    <div
      className="board"
    >
      <div
        className="board-row"
      >
        <button
          className="ac per"
          name="AC"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="per"
          name="+/-"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="per"
          name="%"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="ornge per"
          name="รท"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </div>
      <div
        className="board-row"
      >
        <button
          className="ac"
          name="7"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          name="8"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          name="9"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="ornge"
          name="X"
          onClick={[Function]}
          type="button"
        >
          X
        </button>
      </div>
      <div
        className="board-row"
      >
        <button
          className="ac"
          name="4"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          name="5"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          name="6"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="ornge"
          name="-"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="board-row"
      >
        <button
          className="ac"
          name="1"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          name="2"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          name="3"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="ornge"
          name="+"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="board-row"
      >
        <button
          className="zeros ac"
          name="0"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          name="."
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="ornge"
          name="="
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`App render should render calculator 1`] = `
<div
  className="app"
>
  <div
    className="lets"
  >
    <h1>
      Lets Do Some Magicians with Math Calc! 
    </h1>
  </div>
  <div
    className="calc"
  >
    <div
      className="display-result"
    >
      <input
        defaultValue={null}
        dir="rtl"
        id="result"
        placeholder="0"
      />
    </div>
    <div
      className="board"
    >
      <div
        className="board-row"
      >
        <button
          className="ac per"
          name="AC"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="per"
          name="+/-"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="per"
          name="%"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="ornge per"
          name="รท"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </div>
      <div
        className="board-row"
      >
        <button
          className="ac"
          name="7"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          name="8"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          name="9"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="ornge"
          name="X"
          onClick={[Function]}
          type="button"
        >
          X
        </button>
      </div>
      <div
        className="board-row"
      >
        <button
          className="ac"
          name="4"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          name="5"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          name="6"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="ornge"
          name="-"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="board-row"
      >
        <button
          className="ac"
          name="1"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          name="2"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          name="3"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="ornge"
          name="+"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="board-row"
      >
        <button
          className="zeros ac"
          name="0"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          name="."
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="ornge"
          name="="
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
